src/async_impl/client.rs

```diff
@@ -252,6 +258,9 @@ impl ClientBuilder {
                             HttpVersionPref::Http2 => {
                                 tls.request_alpns(&["h2"]);
                             }
+                            HttpVersionPref::Http3 => {
+                                unreachable!("HTTP/3 shouldn't be enabled unless the feature is")
```

Comment:
We can place `#[cfg(feature = "http3")]` on the enum variant so then this unreachable isn't needed.